File: csharp/FullTimeEmployee.cs
using System;

[Serializable]
public abstract class FullTimeEmployee : Employee
{
    public double BaseSalary { get; set; }
    public string BenefitsPackage { get; set; }

    protected FullTimeEmployee() { }

    protected FullTimeEmployee(string id, string firstName, string lastName, DateTime dob,
        string position, string department, double baseSalary, string benefits)
        : base(id, firstName, lastName, dob, position, department)
    {
        BaseSalary = baseSalary;
        BenefitsPackage = benefits;
    }

    public override double GetAnnualSalary() => BaseSalary;

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} (Full-Time) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} (Full-Time) has no assigned manager.");
    }
}

File: java/FullTimeEmployee.java
import java.util.Date;

public abstract class FullTimeEmployee extends Employee {
    private double baseSalary;
    private String benefitsPackage;

    public FullTimeEmployee(String id, String firstName, String lastName, Date dob,
            String position, String department, double baseSalary, String benefits) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department);
        this.baseSalary = baseSalary;
        this.benefitsPackage = benefits;
    }

    @Override
    public double getAnnualSalary() {
        return baseSalary;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (Full-Time) reports to manager " + manager.getFullName());
        } else {
            System.out.println(getFullName() + " (Full-Time) has no assigned manager.");
        }
    }
}